src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1798,13 +1801,17 @@ bool Arguments::check_vm_args_consistency() {
   status = CompilerConfig::check_args_consistency(status);
 #if INCLUDE_JVMCI
   if (status && EnableJVMCI) {
-    PropertyList_unique_add(&_system_properties, "jdk.internal.vm.ci.enabled", "true",
-        AddProperty, UnwriteableProperty, InternalProperty);
-    if (ClassLoader::is_module_observable("jdk.internal.vm.ci")) {
+    // libjvmci doesn't require resolving jdk.internal.vm.ci as it is
+    // compiled into the libjvmci image itself. Without libjvmci, there
+    // is no other representation of the jdk.internal.vm.ci module
+    // so it needs to be added to the root module set.
+    if (!UseJVMCINativeLibrary && ClassLoader::is_module_observable("jdk.internal.vm.ci") && !_jvmci_module_added) {
       if (!create_numbered_module_property("jdk.module.addmods", "jdk.internal.vm.ci", _addmods_count++)) {
         return false;
       }
     }
+    PropertyList_unique_add(&_system_properties, "jdk.internal.vm.ci.enabled", "true",
+        AddProperty, UnwriteableProperty, InternalProperty);
```

Comment:
What's the purpose of the `jdk.internal.vm.ci.enabled` property? Should it be enabled only if the `jdk.internal.vm.ci` module is added?

Reply:
It exists to [check](https://github.com/search?q=repo%3Aopenjdk%2Fjdk%20checkJVMCIEnabled&type=code) in various Java-level entry points that the JVMCI VM support has been enabled so a nicer error message can be thrown that provides a possible corrective action. However, since it's now impossible to load the JVMCI module without enabling the JVMCI VM support, this is no longer of any use. I'll remove it.